### PR TITLE
CBG-3865: Added 3.1.5 release notes

### PR DESCRIPTION
**Pull request by ElliotFrancisHunter**
PR for the ticket: https://issues.couchbase.com/browse/CBG-3865

Added 3.1.5 release notes and a warning to revert 3.1.4 back to 3.1.3 or upgrade to 3.1.5.

### CODE REVIEW

File: modules/ROOT/pages/_partials/release_notes/sync-gateway-3-1-5-release-note.adoc [hunk 1]
```diff
@@ -0,0 +1,64 @@
+== 3.1.5 -- April 2024
+
+Version 3.1.5 of Sync Gateway delivers the following features and enhancements:
+
+NOTE: For an overview of the latest features offered in Sync Gateway 3.1, see xref:whatsnew.adoc[New in 3.1].
+
+[WARNING]
+--
+
+We have discovered a critical issue impacting a few customers that utilize OIDC.
```
Comment (kevdotpub) [inline modules/ROOT/pages/_partials/release_notes/sync-gateway-3-1-5-release-note.adoc]: Should this be clearer that the issue is with `3.1.4`?

File: modules/ROOT/pages/_partials/release_notes/sync-gateway-3-1-5-release-note.adoc [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+Version 3.1.5 of Sync Gateway delivers the following features and enhancements:
+
+NOTE: For an overview of the latest features offered in Sync Gateway 3.1, see xref:whatsnew.adoc[New in 3.1].
+
+[WARNING]
+--
+
+We have discovered a critical issue impacting a few customers that utilize OIDC. 
+This issue could lead to users losing access to channels, consequently resulting in document revocations.
+We suggest all users revert their Sync Gateway to version 3.1.3 as a precautionary step or promptly upgrade to version 3.1.5.
```
Comment (kevdotpub) [inline modules/ROOT/pages/_partials/release_notes/sync-gateway-3-1-5-release-note.adoc]: Now `3.1.5` is available there's no reason for anyone to downgrade. I think we should only suggest upgrading to `3.1.5`

**Review by kevdotpub — CHANGES_REQUESTED** [review]

File: modules/ROOT/pages/_partials/release_notes/sync-gateway-3-1-5-release-note.adoc (hunk 2 above)
Comment (ElliotFrancisHunter) [reply modules/ROOT/pages/_partials/release_notes/sync-gateway-3-1-5-release-note.adoc]: Okay will change!

File: modules/ROOT/pages/_partials/release_notes/sync-gateway-3-1-5-release-note.adoc (hunk 1 above)
Comment (ElliotFrancisHunter) [reply modules/ROOT/pages/_partials/release_notes/sync-gateway-3-1-5-release-note.adoc]: Can make this clearer yes!

**Review by kevdotpub — APPROVED** [review]
Perfect! :shipit:

**Review by bfavini — APPROVED** [review]

**Pull request merged**